validation.js

```diff
@@ -19,3 +21,23 @@ form.addEventListener('submit', (event) => {
   }
   event.preventDefault();
 });
+
+function saveInfo() {
+  const userInfo = JSON.stringify({
+    username: userName.value,
+    email: email.value,
+    text:
+     text.value,
+  });
+  localStorage.setItem('userInfo', userInfo);
+}
+
+userName.addEventListener('input', saveInfo);
+email.addEventListener('input', saveInfo);
+text.addEventListener('input', saveInfo);
+
+const userInfo = localStorage.getItem('userInfo');
+
+userName.value = JSON.parse(userInfo)?.username ? JSON.parse(userInfo).username : '';
+email.value = JSON.parse(userInfo)?.email ? JSON.parse(userInfo).email : '';
+text.value = JSON.parse(userInfo)?.text ? JSON.parse(userInfo).text : '';
```

Comment:
[OPTIONAL] Some patterns can be seen here 👀 look to see if you can have a helper function whenever there's some recurring pattern. If you need any help implementing it, let me know. 😉